near-sdk/src/lib.rs

```diff
@@ -99,10 +98,330 @@
+/// any arguments.
+///
+/// All fields(version, link) are optional and will be populated with defaults from the Cargo.toml file if not specified.
+/// The `standard` will be populated with `nep330` by default.
+///
+/// Any additional standards can be added and should be specified using the `standard` sub-attribute.
+///
+/// The `contract_source_metadata()` view function will be added and can be used to retrieve the source metadata.
+/// Also, the source metadata will be stored as a constant, `CONTRACT_SOURCE_METADATA`, in the contract code.
+///
+/// ## Examples
+/// ```rust
+/// use near_sdk::near;
+///
+/// #[near(contract_metadata(
+///     version = "39f2d2646f2f60e18ab53337501370dc02a5661c",
+///     link = "https://github.com/near-examples/nft-tutorial",
+///     standard(standard = "nep330", version = "1.1.0"),
+///     standard(standard = "nep171", version = "1.0.0"),
+///     standard(standard = "nep177", version = "2.0.0"),
+/// ))]
+/// struct Contract {}
+/// ```
+pub use near_sdk_macros::near;
+
+/// `ext_contract` takes a Rust Trait and converts it to a module with static methods.
+/// Each of these static methods takes positional arguments defined by the Trait,
+/// then the receiver_id, the attached deposit and the amount of gas and returns a new Promise.
+///
+/// ## Examples
+///
+/// ```rust
+/// use near_sdk::{AccountId,ext_contract, near, Promise, Gas};
+///
+/// #[near(contract_state)]
+/// struct Contract {
+///     calculator_account: AccountId,
+/// }
+///
+/// #[ext_contract(ext_calculator)]
+/// trait Calculator {
+///     fn mult(&self, a: u64, b: u64) -> u128;
+///     fn sum(&self, a: u128, b: u128) -> u128;
+/// }
+///
+/// #[near]
+/// impl Contract {
+///    pub fn multiply_by_five(&mut self, number: u64) -> Promise {
+///        ext_calculator::ext(self.calculator_account.clone())
+///            .with_static_gas(Gas::from_tgas(5))
+///            .mult(number, 5)
+///    }
+/// }
+///
+/// ```
+///
+/// See more information about role of ext_contract in [NEAR documentation](https://docs.near.org/build/smart-contracts/anatomy/crosscontract)
+pub use near_sdk_macros::ext_contract;
+
+/// This macro is deprecated. Use [macro@near] instead. The difference between `#[near]` and `#[near_bindgen]` is that
+/// with `#[near_bindgen]` you have to manually add boilerplate code for structs and enums so that they become Json- and Borsh-serializable:
+/// ```rust
+/// use near_sdk::{near_bindgen, NearSchema, borsh::{BorshSerialize, BorshDeserialize}};
+///
+/// #[near_bindgen]
+/// #[derive(BorshSerialize, BorshDeserialize, NearSchema)]
+/// #[borsh(crate = "near_sdk::borsh")]
+/// struct MyStruct {
+///    pub name: String,
+/// }
+/// ```
+/// Instead of:
+/// ```rust
+/// use near_sdk::near;
+///
+/// #[near(serializers=[borsh])]
+/// struct MyStruct {
+///     pub name: String,
+/// }
+/// ```
+pub use near_sdk_macros::near_bindgen;
```

Comment:
`near_bindgen` can be cut and pasted just after `near`, because they're both twin brothers, one handsome, another - an ugly one